crypto/core_namemap.c

```diff
@@ -107,7 +107,8 @@ int ossl_namemap_empty(OSSL_NAMEMAP *namemap)
     if (namemap == NULL)
         return 1;
 
-    CRYPTO_THREAD_read_lock(namemap->lock);
+    if (!CRYPTO_THREAD_read_lock(namemap->lock))
+        return 0;
```

Comment:
I'm not sure 0 is a good way to indicate an error.

Reply:
> I'm not sure 0 is a good way to indicate an error.

but consistent with how the function is written.

Reply:
I'm not sure it is, I think the usage in ossl_namemap_stored() could use an error case.

Reply:
Tell me what value you would like to see, @kroeckx,  and I'll make the change.  And then you can fight with @paulidale :)

Reply:
@levitte wrote this, but I suggest: _e_<sup>&#120102;&#960;</sup>

Reply:
Works for me.

Reply:
okay -1 it is.